Failed to compile `zlib-ng` on armv7 and aarch64 linux

Compiling `zlib-ng` on `armv7-unknown-linux-musleabihf` failed with:

<details>

```
  just check
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
    CARGO_BUILD_TARGET: armv7-unknown-linux-musleabihf
    tools: just,cargo-hack
    CACHE_ON_FAILURE: false
    CARGO_INCREMENTAL: 0
    JUST_USE_CARGO_ZIGBUILD: true
env RUSTFLAGS='', CARGO='cargo-zigbuild'
cargo-zigbuild check  --target armv7-unknown-linux-musleabihf --no-default-features --features git,rustls,fancy-with-backtrace,zstd-thin,log_max_level_debug
warning: /home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-manifests/Cargo.toml: unused manifest key: workaround-avoid-dev-deps
warning: /home/runner/work/cargo-binstall/cargo-binstall/crates/detect-targets/Cargo.toml: unused manifest key: workaround-avoid-dev-deps
    Updating crates.io index
 Downloading crates ...
  Downloaded sha1 v0.10.5
  Downloaded sha1-asm v0.5.1
  Downloaded cpufeatures v0.2.8
   Compiling cmake v0.1.50
    Checking crypto-common v0.1.6
    Checking digest v0.10.7
    Checking sha1 v0.10.5
    Checking binstalk-types v0.5.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-types)
   Compiling detect-targets v0.1.8 (/home/runner/work/cargo-binstall/cargo-binstall/crates/detect-targets)
   Compiling libz-ng-sys v1.1.9
   Compiling leon v2.0.1 (/home/runner/work/cargo-binstall/cargo-binstall/crates/leon)
    Checking fs-lock v0.1.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/fs-lock)
   Compiling leon-macros v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/leon-macros)
   Compiling cargo-binstall v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/bin)
    Checking normalize-path v0.2.1 (/home/runner/work/cargo-binstall/cargo-binstall/crates/normalize-path)
    Checking binstalk-manifests v0.6.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-manifests)
    Checking detect-wasi v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/detect-wasi)
error: failed to run custom build command for `libz-ng-sys v1.1.9`

Caused by:
  process didn't exit successfully: `/home/runner/work/cargo-binstall/cargo-binstall/target/debug/build/libz-ng-sys-e42ac8d6410a633c/build-script-build_zng` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_armv7-unknown-linux-musleabihf = None
  CMAKE_TOOLCHAIN_FILE_armv7_unknown_linux_musleabihf = Some("/home/runner/.cache/cargo-zigbuild/0.16.11/cmake/armv7-unknown-linux-musleabihf-toolchain.cmake")
  CMAKE_GENERATOR_armv7-unknown-linux-musleabihf = None
  CMAKE_GENERATOR_armv7_unknown_linux_musleabihf = None
  TARGET_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_armv7-unknown-linux-musleabihf = None
  CMAKE_PREFIX_PATH_armv7_unknown_linux_musleabihf = None
  TARGET_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_armv7-unknown-linux-musleabihf = None
  CMAKE_armv7_unknown_linux_musleabihf = None
  TARGET_CMAKE = None
  CMAKE = None
  running: cd "/home/runner/work/cargo-binstall/cargo-binstall/target/armv7-unknown-linux-musleabihf/debug/build/libz-ng-sys-c0194d3cfb129cb9/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng" "-DBUILD_SHARED_LIBS=OFF" "-DZLIB_COMPAT=OFF" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_TOOLCHAIN_FILE=/home/runner/.cache/cargo-zigbuild/0.16.11/cmake/armv7-unknown-linux-musleabihf-toolchain.cmake" "-DCMAKE_INSTALL_PREFIX=/home/runner/work/cargo-binstall/cargo-binstall/target/armv7-unknown-linux-musleabihf/debug/build/libz-ng-sys-c0194d3cfb129cb9/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC -march=armv7-a -mfpu=vfpv3-d16" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC -march=armv7-a -mfpu=vfpv3-d16" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC -march=armv7-a -mfpu=vfpv3-d16" "-DCMAKE_BUILD_TYPE=Debug"
  -- Using CMake version 3.26.4
  -- ZLIB_HEADER_VERSION: 1.2.11
  -- ZLIBNG_HEADER_VERSION: 2.0.7
  -- The C compiler identification is Clang 15.0.7
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /home/runner/.cache/cargo-zigbuild/0.16.11/zigcc-armv7-unknown-linux-musleabihf.sh - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Arch not recognized, falling back to cmake arch: 'armv7'
  -- Basearch of 'armv7' has been detected as: 'arm'
  -- Using CMake toolchain: /home/runner/.cache/cargo-zigbuild/0.16.11/cmake/armv7-unknown-linux-musleabihf-toolchain.cmake
  -- Performing Test HAVE_FLOATABI_SOFTFP
  -- Performing Test HAVE_FLOATABI_SOFTFP - Success
  -- ARM floating point arch: -mfloat-abi=softfp
  -- Performing Test MFPU_NEON_AVAILABLE
  -- Performing Test MFPU_NEON_AVAILABLE - Success
  -- Performing Test FNO_LTO_AVAILABLE
  -- Performing Test FNO_LTO_AVAILABLE - Success
  -- Looking for sys/sdt.h
  -- Looking for sys/sdt.h - not found
  -- Looking for unistd.h
  -- Looking for unistd.h - found
  -- Looking for sys/types.h
  -- Looking for sys/types.h - found
  -- Looking for stdint.h
  -- Looking for stdint.h - found
  -- Looking for stddef.h
  -- Looking for stddef.h - found
  -- Check size of off64_t
  -- Check size of off64_t - done
  -- Looking for fseeko
  -- Looking for fseeko - found
  -- Looking for strerror
  -- Looking for strerror - found
  -- Looking for posix_memalign
  -- Looking for posix_memalign - found
  -- Looking for aligned_alloc
  -- Looking for aligned_alloc - found
  -- Performing Test HAVE_NO_INTERPOSITION
  -- Performing Test HAVE_NO_INTERPOSITION - Success
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN - Success
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL - Success
  -- Performing Test HAVE_BUILTIN_CTZ
  -- Performing Test HAVE_BUILTIN_CTZ - Success
  -- Performing Test HAVE_BUILTIN_CTZLL
  -- Performing Test HAVE_BUILTIN_CTZLL - Success
  -- Performing Test HAVE_PTRDIFF_T
  -- Performing Test HAVE_PTRDIFF_T - Success
  -- Performing Test ARM_AUXV_HAS_CRC32
  -- Performing Test ARM_AUXV_HAS_CRC32 - Success
  -- Performing Test ARM_AUXV_HAS_NEON
  -- Performing Test ARM_AUXV_HAS_NEON - Success
  -- Architecture-specific source files: arch/arm/armfeature.c;arch/arm/adler32_neon.c;arch/arm/chunkset_neon.c;arch/arm/slide_neon.c
  -- The following features have been enabled:

   * CMAKE_BUILD_TYPE, Build type: Debug (selected)
   * WITH_GZFILEOP, Compile with support for gzFile related functions
   * WITH_OPTIM, Build with optimisation
   * WITH_NEW_STRATEGIES, Use new strategies
   * WITH_UNALIGNED, Support unaligned reads on platforms that support it
   * WITH_ACLE, Build with ACLE
   * WITH_NEON, Build with NEON intrinsics
   * NEON_ADLER32, Support NEON instructions in adler32, using "-mfpu=neon"
   * NEON_SLIDEHASH, Support NEON instructions in slide_hash, using "-mfpu=neon"

  -- The following features have been disabled:

   * ZLIB_COMPAT, Compile with zlib compatible API
   * ZLIB_ENABLE_TESTS, Build test binaries
   * ZLIB_DUAL_LINK, Dual link tests against system zlib
   * WITH_FUZZERS, Build test/fuzz
   * WITH_NATIVE_INSTRUCTIONS, Instruct the compiler to use the full instruction set on this host (gcc/clang -march=native)
   * WITH_MAINTAINER_WARNINGS, Build with project maintainer warnings
   * WITH_CODE_COVERAGE, Enable code coverage reporting
   * WITH_INFLATE_STRICT, Build with strict inflate distance checking
   * WITH_INFLATE_ALLOW_INVALID_DIST, Build with zero fill for inflate invalid distances
   * INSTALL_UTILS, Copy minigzip and minideflate during install

  -- Configuring done (34.1s)
  -- Generating done (0.0s)
  -- Build files have been written to: /home/runner/work/cargo-binstall/cargo-binstall/target/armv7-unknown-linux-musleabihf/debug/build/libz-ng-sys-c0194d3cfb129cb9/out/build
  running: cd "/home/runner/work/cargo-binstall/cargo-binstall/target/armv7-unknown-linux-musleabihf/debug/build/libz-ng-sys-c0194d3cfb129cb9/out/build" && MAKEFLAGS="-j --jobserver-fds=8,11 --jobserver-auth=8,11" "cmake" "--build" "." "--target" "install" "--config" "Debug"
  [  3%] Building C object CMakeFiles/zlib.dir/adler32.c.o
  [  7%] Building C object CMakeFiles/zlib.dir/chunkset.c.o
  [ 10%] Building C object CMakeFiles/zlib.dir/compare258.c.o
  [ 14%] Building C object CMakeFiles/zlib.dir/compress.c.o
  [ 17%] Building C object CMakeFiles/zlib.dir/crc32.c.o
  [ 21%] Building C object CMakeFiles/zlib.dir/crc32_comb.c.o
  [ 25%] Building C object CMakeFiles/zlib.dir/deflate.c.o
  [ 28%] Building C object CMakeFiles/zlib.dir/deflate_fast.c.o
  [ 32%] Building C object CMakeFiles/zlib.dir/deflate_medium.c.o
  [ 35%] Building C object CMakeFiles/zlib.dir/deflate_quick.c.o
  [ 39%] Building C object CMakeFiles/zlib.dir/deflate_slow.c.o
  [ 42%] Building C object CMakeFiles/zlib.dir/functable.c.o
  [ 46%] Building C object CMakeFiles/zlib.dir/infback.c.o
  [ 50%] Building C object CMakeFiles/zlib.dir/inffast.c.o
  [ 53%] Building C object CMakeFiles/zlib.dir/inflate.c.o
  [ 57%] Building C object CMakeFiles/zlib.dir/inftrees.c.o
  [ 60%] Building C object CMakeFiles/zlib.dir/insert_string.c.o
  [ 64%] Building C object CMakeFiles/zlib.dir/trees.c.o
  [ 67%] Building C object CMakeFiles/zlib.dir/uncompr.c.o
  [ 71%] Building C object CMakeFiles/zlib.dir/zutil.c.o
  [ 75%] Building C object CMakeFiles/zlib.dir/arch/arm/armfeature.c.o
  [ 78%] Building C object CMakeFiles/zlib.dir/arch/arm/adler32_neon.c.o
  [ 82%] Building C object CMakeFiles/zlib.dir/arch/arm/chunkset_neon.c.o

  --- stderr
  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_FLAGS
      CMAKE_CXX_FLAGS


  In file included from /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:11:
  /home/runner/.local/lib/python3.10/site-packages/ziglang/lib/include/arm_neon.h:32:2: error: "NEON support not enabled"
  #error "NEON support not enabled"
   ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:23:5: error: use of undeclared identifier 'uint32x2_t'
      uint32x2_t adacc2, s2acc2, as;
      ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:24:5: error: use of undeclared identifier 'uint8x16_t'
      uint8x16_t t0 = vld1q_u8(taps), t1 = vld1q_u8(taps + 16);
      ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:26:5: error: use of undeclared identifier 'uint32x4_t'
      uint32x4_t adacc = vdupq_n_u32(0), s2acc = vdupq_n_u32(0);
      ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:27:5: error: use of undeclared identifier 'adacc'
      adacc = vsetq_lane_u32(s[0], adacc, 0);
      ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:27:13: warning: call to undeclared function 'vsetq_lane_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      adacc = vsetq_lane_u32(s[0], adacc, 0);
              ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:27:34: error: use of undeclared identifier 'adacc'
      adacc = vsetq_lane_u32(s[0], adacc, 0);
                                   ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:28:5: error: use of undeclared identifier 's2acc'
      s2acc = vsetq_lane_u32(s[1], s2acc, 0);
      ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:28:34: error: use of undeclared identifier 's2acc'
      s2acc = vsetq_lane_u32(s[1], s2acc, 0);
                                   ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:31:9: error: use of undeclared identifier 'uint8x16_t'
          uint8x16_t d0 = vld1q_u8(buf), d1 = vld1q_u8(buf + 16);
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:32:9: error: use of undeclared identifier 'uint16x8_t'
          uint16x8_t adler, sum2;
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:33:9: error: use of undeclared identifier 's2acc'
          s2acc = vaddq_u32(s2acc, vshlq_n_u32(adacc, 5));
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:33:17: warning: call to undeclared function 'vaddq_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          s2acc = vaddq_u32(s2acc, vshlq_n_u32(adacc, 5));
                  ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:33:27: error: use of undeclared identifier 's2acc'
          s2acc = vaddq_u32(s2acc, vshlq_n_u32(adacc, 5));
                            ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:33:34: warning: call to undeclared function 'vshlq_n_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          s2acc = vaddq_u32(s2acc, vshlq_n_u32(adacc, 5));
                                   ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:33:46: error: use of undeclared identifier 'adacc'
          s2acc = vaddq_u32(s2acc, vshlq_n_u32(adacc, 5));
                                               ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:34:9: error: use of undeclared identifier 'adler'
          adler = vpaddlq_u8(       d0);
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:34:17: warning: call to undeclared function 'vpaddlq_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          adler = vpaddlq_u8(       d0);
                  ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:34:35: error: use of undeclared identifier 'd0'
          adler = vpaddlq_u8(       d0);
                                    ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:35:9: error: use of undeclared identifier 'adler'
          adler = vpadalq_u8(adler, d1);
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:35:17: warning: call to undeclared function 'vpadalq_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          adler = vpadalq_u8(adler, d1);
                  ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:35:28: error: use of undeclared identifier 'adler'
          adler = vpadalq_u8(adler, d1);
                             ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:35:35: error: use of undeclared identifier 'd1'
          adler = vpadalq_u8(adler, d1);
                                    ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:36:9: error: use of undeclared identifier 'sum2'
          sum2 = vmull_u8(      vget_low_u8(t0), vget_low_u8(d0));
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:36:16: warning: call to undeclared function 'vmull_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          sum2 = vmull_u8(      vget_low_u8(t0), vget_low_u8(d0));
                 ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/adler32_neon.c:36:31: warning: call to undeclared function 'vget_low_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
          sum2 = vmull_u8(      vget_low_u8(t0), vget_low_u8(d0));
                                ^
  fatal error: too many errors emitted, stopping now [-ferror-limit=]
  7 warnings and 20 errors generated.
  gmake[2]: *** [CMakeFiles/zlib.dir/build.make:370: CMakeFiles/zlib.dir/arch/arm/adler32_neon.c.o] Error 1
  gmake[2]: *** Waiting for unfinished jobs....
  In file included from /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:9:
  /home/runner/.local/lib/python3.10/site-packages/ziglang/lib/include/arm_neon.h:32:2: error: "NEON support not enabled"
  #error "NEON support not enabled"
   ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:14:9: error: unknown type name 'uint8x16_t'
  typedef uint8x16_t chunk_t;
          ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:25:14: warning: call to undeclared function 'vreinterpretq_u8_u16'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u16(vdupq_n_u16(tmp));
               ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:25:35: warning: call to undeclared function 'vdupq_n_u16'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u16(vdupq_n_u16(tmp));
                                    ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:31:14: warning: call to undeclared function 'vreinterpretq_u8_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u32(vdupq_n_u32(tmp));
               ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:31:35: warning: call to undeclared function 'vdupq_n_u32'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u32(vdupq_n_u32(tmp));
                                    ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:37:14: warning: call to undeclared function 'vreinterpretq_u8_u64'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u64(vdupq_n_u64(tmp));
               ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:37:35: warning: call to undeclared function 'vdupq_n_u64'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vreinterpretq_u8_u64(vdupq_n_u64(tmp));
                                    ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:48:14: warning: call to undeclared function 'vld1q_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      *chunk = vld1q_u8(s);
               ^
  /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng/arch/arm/chunkset_neon.c:52:5: warning: call to undeclared function 'vst1q_u8'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
      vst1q_u8(out, *chunk);
      ^
  8 warnings and 2 errors generated.
  gmake[2]: *** [CMakeFiles/zlib.dir/build.make:384: CMakeFiles/zlib.dir/arch/arm/chunkset_neon.c.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/zlib.dir/all] Error 2
  gmake: *** [Makefile:136: all] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5
```

</details>

On `aarch64-unknown-linux-musl`:

<details>

```
  just check
  shell: /usr/bin/bash -e {0}
  env:
    CARGO_TERM_COLOR: always
    CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
    CARGO_BUILD_TARGET: aarch64-unknown-linux-musl
    tools: just,cargo-hack
    CACHE_ON_FAILURE: false
    CARGO_INCREMENTAL: 0
    JUST_USE_CARGO_ZIGBUILD: true
env RUSTFLAGS='', CARGO='cargo-zigbuild'
cargo-zigbuild check  --target aarch64-unknown-linux-musl --no-default-features --features git,rustls,fancy-with-backtrace,zstd-thin,log_max_level_debug
warning: /home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-manifests/Cargo.toml: unused manifest key: workaround-avoid-dev-deps
warning: /home/runner/work/cargo-binstall/cargo-binstall/crates/detect-targets/Cargo.toml: unused manifest key: workaround-avoid-dev-deps
    Updating crates.io index
 Downloading crates ...
  Downloaded sha1 v0.10.5
  Downloaded sha1-asm v0.5.1
  Downloaded cpufeatures v0.2.8
   Compiling cmake v0.1.50
    Checking crypto-common v0.1.6
   Compiling sha1-asm v0.5.1
    Checking digest v0.10.7
   Compiling libz-ng-sys v1.1.9
    Checking cpufeatures v0.2.8
    Checking binstalk-types v0.5.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-types)
   Compiling detect-targets v0.1.8 (/home/runner/work/cargo-binstall/cargo-binstall/crates/detect-targets)
   Compiling leon v2.0.1 (/home/runner/work/cargo-binstall/cargo-binstall/crates/leon)
    Checking fs-lock v0.1.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/fs-lock)
   Compiling leon-macros v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/leon-macros)
   Compiling cargo-binstall v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/bin)
    Checking normalize-path v0.2.1 (/home/runner/work/cargo-binstall/cargo-binstall/crates/normalize-path)
    Checking binstalk-manifests v0.6.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/binstalk-manifests)
    Checking detect-wasi v1.0.0 (/home/runner/work/cargo-binstall/cargo-binstall/crates/detect-wasi)
The following warnings were emitted during compilation:

warning: info: available CPUs for architecture 'aarch64':
warning:  a64fx
warning:  ampere1
warning:  apple_a10
warning:  apple_a11
warning:  apple_a12
warning:  apple_a13
warning:  apple_a14
warning:  apple_a7
warning:  apple_a8
warning:  apple_a9
warning:  apple_latest
warning:  apple_m1
warning:  apple_s4
warning:  apple_s5
warning:  carmel
warning:  cortex_a34
warning:  cortex_a35
warning:  cortex_a510
warning:  cortex_a53
warning:  cortex_a55
warning:  cortex_a57
warning:  cortex_a65
warning:  cortex_a65ae
warning:  cortex_a710
warning:  cortex_a72
warning:  cortex_a73
warning:  cortex_a75
warning:  cortex_a76
warning:  cortex_a76ae
warning:  cortex_a77
warning:  cortex_a78
warning:  cortex_a78c
warning:  cortex_r82
warning:  cortex_x1
warning:  cortex_x1c
warning:  cortex_x2
warning:  cyclone
warning:  emag
warning:  exynos_m1
warning:  exynos_m2
warning:  exynos_m3
warning:  exynos_m4
warning:  exynos_m5
warning:  falkor
warning:  generic
warning:  kryo
warning:  neoverse_512tvb
warning:  neoverse_e1
warning:  neoverse_n1
warning:  neoverse_n2
warning:  neoverse_v1
warning:  saphira
warning:  thunderx
warning:  thunderx2t99
warning:  thunderx3t110
warning:  thunderxt81
warning:  thunderxt83
warning:  thunderxt88
warning:  tsv110
warning:  xgene1
warning: 
warning: error: unknown CPU: 'armv8'

error: failed to run custom build command for `sha1-asm v0.5.1`

Caused by:
  process didn't exit successfully: `/home/runner/work/cargo-binstall/cargo-binstall/target/debug/build/sha1-asm-2b3c3458edba774d/build-script-build` (exit status: 1)
  --- stdout
  TARGET = Some("aarch64-unknown-linux-musl")
  OPT_LEVEL = Some("0")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CC_aarch64-unknown-linux-musl
  CC_aarch64-unknown-linux-musl = None
  cargo:rerun-if-env-changed=CC_aarch64_unknown_linux_musl
  CC_aarch64_unknown_linux_musl = Some("/home/runner/.cache/cargo-zigbuild/0.16.11/zigcc-aarch64-unknown-linux-musl.sh")
  cargo:rerun-if-env-changed=CFLAGS_aarch64-unknown-linux-musl
  CFLAGS_aarch64-unknown-linux-musl = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_unknown_linux_musl
  CFLAGS_aarch64_unknown_linux_musl = None
  cargo:rerun-if-env-changed=TARGET_CFLAGS
  TARGET_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("neon")
  running: "/home/runner/.cache/cargo-zigbuild/0.16.11/zigcc-aarch64-unknown-linux-musl.sh" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-4" "-fno-omit-frame-pointer" "-Wall" "-Wextra" "-march=armv8-a+crypto" "-c" "-o" "/home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/sha1-asm-0e2e0b5ba6f608c4/out/src/aarch64.o" "-c" "src/aarch64.S"
  cargo:warning=info: available CPUs for architecture 'aarch64':
  cargo:warning= a64fx
  cargo:warning= ampere1
  cargo:warning= apple_a10
  cargo:warning= apple_a11
  cargo:warning= apple_a12
  cargo:warning= apple_a13
  cargo:warning= apple_a14
  cargo:warning= apple_a7
  cargo:warning= apple_a8
  cargo:warning= apple_a9
  cargo:warning= apple_latest
  cargo:warning= apple_m1
  cargo:warning= apple_s4
  cargo:warning= apple_s5
  cargo:warning= carmel
  cargo:warning= cortex_a34
  cargo:warning= cortex_a35
  cargo:warning= cortex_a510
  cargo:warning= cortex_a53
  cargo:warning= cortex_a55
  cargo:warning= cortex_a57
  cargo:warning= cortex_a65
  cargo:warning= cortex_a65ae
  cargo:warning= cortex_a710
  cargo:warning= cortex_a72
  cargo:warning= cortex_a73
  cargo:warning= cortex_a75
  cargo:warning= cortex_a76
  cargo:warning= cortex_a76ae
  cargo:warning= cortex_a77
  cargo:warning= cortex_a78
  cargo:warning= cortex_a78c
  cargo:warning= cortex_r82
  cargo:warning= cortex_x1
  cargo:warning= cortex_x1c
  cargo:warning= cortex_x2
  cargo:warning= cyclone
  cargo:warning= emag
  cargo:warning= exynos_m1
  cargo:warning= exynos_m2
  cargo:warning= exynos_m3
  cargo:warning= exynos_m4
  cargo:warning= exynos_m5
  cargo:warning= falkor
  cargo:warning= generic
  cargo:warning= kryo
  cargo:warning= neoverse_512tvb
  cargo:warning= neoverse_e1
  cargo:warning= neoverse_n1
  cargo:warning= neoverse_n2
  cargo:warning= neoverse_v1
  cargo:warning= saphira
  cargo:warning= thunderx
  cargo:warning= thunderx2t99
  cargo:warning= thunderx3t110
  cargo:warning= thunderxt81
  cargo:warning= thunderxt83
  cargo:warning= thunderxt88
  cargo:warning= tsv110
  cargo:warning= xgene1
  cargo:warning=
  cargo:warning=error: unknown CPU: 'armv8'
  exit status: 1

  --- stderr


  error occurred: Command "/home/runner/.cache/cargo-zigbuild/0.16.11/zigcc-aarch64-unknown-linux-musl.sh" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-gdwarf-4" "-fno-omit-frame-pointer" "-Wall" "-Wextra" "-march=armv8-a+crypto" "-c" "-o" "/home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/sha1-asm-0e2e0b5ba6f608c4/out/src/aarch64.o" "-c" "src/aarch64.S" with args "zigcc-aarch64-unknown-linux-musl.sh" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `libz-ng-sys v1.1.9`

Caused by:
  process didn't exit successfully: `/home/runner/work/cargo-binstall/cargo-binstall/target/debug/build/libz-ng-sys-e42ac8d6410a633c/build-script-build_zng` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_aarch64-unknown-linux-musl = None
  CMAKE_TOOLCHAIN_FILE_aarch64_unknown_linux_musl = Some("/home/runner/.cache/cargo-zigbuild/0.16.11/cmake/aarch64-unknown-linux-musl-toolchain.cmake")
  CMAKE_GENERATOR_aarch64-unknown-linux-musl = None
  CMAKE_GENERATOR_aarch64_unknown_linux_musl = None
  TARGET_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_aarch64-unknown-linux-musl = None
  CMAKE_PREFIX_PATH_aarch64_unknown_linux_musl = None
  TARGET_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_aarch64-unknown-linux-musl = None
  CMAKE_aarch64_unknown_linux_musl = None
  TARGET_CMAKE = None
  CMAKE = None
  running: cd "/home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/libz-ng-sys-8e7ae9e5b16a1bc5/out/build" && CMAKE_PREFIX_PATH="" "cmake" "/home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libz-ng-sys-1.1.9/src/zlib-ng" "-DBUILD_SHARED_LIBS=OFF" "-DZLIB_COMPAT=OFF" "-DZLIB_ENABLE_TESTS=OFF" "-DWITH_GZFILEOP=ON" "-DCMAKE_TOOLCHAIN_FILE=/home/runner/.cache/cargo-zigbuild/0.16.11/cmake/aarch64-unknown-linux-musl-toolchain.cmake" "-DCMAKE_INSTALL_PREFIX=/home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/libz-ng-sys-8e7ae9e5b16a1bc5/out" "-DCMAKE_C_FLAGS= -ffunction-sections -fdata-sections -fPIC" "-DCMAKE_CXX_FLAGS= -ffunction-sections -fdata-sections -fPIC" "-DCMAKE_ASM_FLAGS= -ffunction-sections -fdata-sections -fPIC" "-DCMAKE_BUILD_TYPE=Debug"
  -- Using CMake version 3.26.4
  -- ZLIB_HEADER_VERSION: 1.2.11
  -- ZLIBNG_HEADER_VERSION: 2.0.7
  -- The C compiler identification is Clang 15.0.7
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /home/runner/.cache/cargo-zigbuild/0.16.11/zigcc-aarch64-unknown-linux-musl.sh - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Arch not recognized, falling back to cmake arch: 'aarch64'
  -- Basearch of 'aarch64' has been detected as: 'arm'
  -- Using CMake toolchain: /home/runner/.cache/cargo-zigbuild/0.16.11/cmake/aarch64-unknown-linux-musl-toolchain.cmake
  -- Performing Test FNO_LTO_AVAILABLE
  -- Performing Test FNO_LTO_AVAILABLE - Success
  -- Architecture supports unaligned reads
  -- Architecture supports unaligned reads of > 4 bytes
  -- Looking for sys/sdt.h
  -- Looking for sys/sdt.h - not found
  -- Looking for unistd.h
  -- Looking for unistd.h - found
  -- Looking for sys/types.h
  -- Looking for sys/types.h - found
  -- Looking for stdint.h
  -- Looking for stdint.h - found
  -- Looking for stddef.h
  -- Looking for stddef.h - found
  -- Check size of off64_t
  -- Check size of off64_t - done
  -- Looking for fseeko
  -- Looking for fseeko - found
  -- Looking for strerror
  -- Looking for strerror - found
  -- Looking for posix_memalign
  -- Looking for posix_memalign - found
  -- Looking for aligned_alloc
  -- Looking for aligned_alloc - found
  -- Performing Test HAVE_NO_INTERPOSITION
  -- Performing Test HAVE_NO_INTERPOSITION - Success
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN - Success
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL
  -- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL - Success
  -- Performing Test HAVE_BUILTIN_CTZ
  -- Performing Test HAVE_BUILTIN_CTZ - Success
  -- Performing Test HAVE_BUILTIN_CTZLL
  -- Performing Test HAVE_BUILTIN_CTZLL - Success
  -- Performing Test HAVE_PTRDIFF_T
  -- Performing Test HAVE_PTRDIFF_T - Success
  -- Performing Test ARM_AUXV_HAS_CRC32
  -- Performing Test ARM_AUXV_HAS_CRC32 - Success
  -- Architecture-specific source files: arch/arm/armfeature.c;arch/arm/crc32_acle.c;arch/arm/insert_string_acle.c;arch/arm/adler32_neon.c;arch/arm/chunkset_neon.c;arch/arm/slide_neon.c
  -- The following features have been enabled:

   * CMAKE_BUILD_TYPE, Build type: Debug (selected)
   * WITH_GZFILEOP, Compile with support for gzFile related functions
   * WITH_OPTIM, Build with optimisation
   * WITH_NEW_STRATEGIES, Use new strategies
   * WITH_UNALIGNED, Support unaligned reads on platforms that support it
   * WITH_ACLE, Build with ACLE
   * WITH_NEON, Build with NEON intrinsics
   * ACLE_CRC, Support ACLE optimized CRC hash generation, using "-march=armv8-a+crc"
   * NEON_ADLER32, Support NEON instructions in adler32, using "-march=armv8-a+simd"
   * NEON_SLIDEHASH, Support NEON instructions in slide_hash, using "-march=armv8-a+simd"

  -- The following features have been disabled:

   * ZLIB_COMPAT, Compile with zlib compatible API
   * ZLIB_ENABLE_TESTS, Build test binaries
   * ZLIB_DUAL_LINK, Dual link tests against system zlib
   * WITH_FUZZERS, Build test/fuzz
   * WITH_NATIVE_INSTRUCTIONS, Instruct the compiler to use the full instruction set on this host (gcc/clang -march=native)
   * WITH_MAINTAINER_WARNINGS, Build with project maintainer warnings
   * WITH_CODE_COVERAGE, Enable code coverage reporting
   * WITH_INFLATE_STRICT, Build with strict inflate distance checking
   * WITH_INFLATE_ALLOW_INVALID_DIST, Build with zero fill for inflate invalid distances
   * INSTALL_UTILS, Copy minigzip and minideflate during install

  -- Configuring done (30.7s)
  -- Generating done (0.0s)
  -- Build files have been written to: /home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/libz-ng-sys-8e7ae9e5b16a1bc5/out/build
  running: cd "/home/runner/work/cargo-binstall/cargo-binstall/target/aarch64-unknown-linux-musl/debug/build/libz-ng-sys-8e7ae9e5b16a1bc5/out/build" && MAKEFLAGS="-j --jobserver-fds=8,11 --jobserver-auth=8,11" "cmake" "--build" "." "--target" "install" "--config" "Debug"
  [  3%] Building C object CMakeFiles/zlib.dir/adler32.c.o
  [  6%] Building C object CMakeFiles/zlib.dir/compare258.c.o
  [ 10%] Building C object CMakeFiles/zlib.dir/chunkset.c.o
  [ 13%] Building C object CMakeFiles/zlib.dir/compress.c.o
  [ 16%] Building C object CMakeFiles/zlib.dir/crc32.c.o
  [ 20%] Building C object CMakeFiles/zlib.dir/crc32_comb.c.o
  [ 23%] Building C object CMakeFiles/zlib.dir/deflate.c.o
  [ 26%] Building C object CMakeFiles/zlib.dir/deflate_fast.c.o
  [ 30%] Building C object CMakeFiles/zlib.dir/deflate_medium.c.o
  [ 33%] Building C object CMakeFiles/zlib.dir/deflate_quick.c.o
  [ 36%] Building C object CMakeFiles/zlib.dir/deflate_slow.c.o
  [ 40%] Building C object CMakeFiles/zlib.dir/functable.c.o
  [ 43%] Building C object CMakeFiles/zlib.dir/infback.c.o
  [ 46%] Building C object CMakeFiles/zlib.dir/inffast.c.o
  [ 50%] Building C object CMakeFiles/zlib.dir/inflate.c.o
  [ 53%] Building C object CMakeFiles/zlib.dir/insert_string.c.o
  [ 56%] Building C object CMakeFiles/zlib.dir/inftrees.c.o
  [ 60%] Building C object CMakeFiles/zlib.dir/trees.c.o
  [ 63%] Building C object CMakeFiles/zlib.dir/uncompr.c.o
  [ 66%] Building C object CMakeFiles/zlib.dir/zutil.c.o
  [ 70%] Building C object CMakeFiles/zlib.dir/arch/arm/armfeature.c.o
  [ 73%] Building C object CMakeFiles/zlib.dir/arch/arm/crc32_acle.c.o
  [ 76%] Building C object CMakeFiles/zlib.dir/arch/arm/insert_string_acle.c.o
  [ 80%] Building C object CMakeFiles/zlib.dir/arch/arm/adler32_neon.c.o

  --- stderr
  CMake Warning:
    Manually-specified variables were not used by the project:

      CMAKE_ASM_FLAGS
      CMAKE_CXX_FLAGS


  info: available CPUs for architecture 'aarch64':
   a64fx
   ampere1
   apple_a10
   apple_a11
   apple_a12
   apple_a13
   apple_a14
   apple_a7
   apple_a8
   apple_a9
   apple_latest
   apple_m1
   apple_s4
   apple_s5
   carmel
   cortex_a34
   cortex_a35
   cortex_a510
   cortex_a53
   cortex_a55
   cortex_a57
   cortex_a65
   cortex_a65ae
   cortex_a710
   cortex_a72
   cortex_a73
   cortex_a75
   cortex_a76
   cortex_a76ae
   cortex_a77
   cortex_a78
   cortex_a78c
   cortex_r82
   cortex_x1
   cortex_x1c
   cortex_x2
   cyclone
   emag
   exynos_m1
   exynos_m2
   exynos_m3
   exynos_m4
   exynos_m5
   falkor
   generic
   kryo
   neoverse_512tvb
   neoverse_e1
   neoverse_n1
   neoverse_n2
   neoverse_v1
   saphira
   thunderx
   thunderx2t99
   thunderx3t110
   thunderxt81
   thunderxt83
   thunderxt88
   tsv110
   xgene1

  error: unknown CPU: 'armv8'
  gmake[2]: *** [CMakeFiles/zlib.dir/build.make:370: CMakeFiles/zlib.dir/arch/arm/crc32_acle.c.o] Error 1
  gmake[2]: *** Waiting for unfinished jobs....
  info: available CPUs for architecture 'aarch64':
   a64fx
   ampere1
   apple_a10
   apple_a11
   apple_a12
   apple_a13
   apple_a14
   apple_a7
   apple_a8
   apple_a9
   apple_latest
   apple_m1
   apple_s4
   apple_s5
   carmel
   cortex_a34
   cortex_a35
   cortex_a510
   cortex_a53
   cortex_a55
   cortex_a57
   cortex_a65
   cortex_a65ae
   cortex_a710
   cortex_a72
   cortex_a73
   cortex_a75
   cortex_a76
   cortex_a76ae
   cortex_a77
   cortex_a78
   cortex_a78c
   cortex_r82
   cortex_x1
   cortex_x1c
   cortex_x2
   cyclone
   emag
   exynos_m1
   exynos_m2
   exynos_m3
   exynos_m4
   exynos_m5
   falkor
   generic
   kryo
   neoverse_512tvb
   neoverse_e1
   neoverse_n1
   neoverse_n2
   neoverse_v1
   saphira
   thunderx
   thunderx2t99
   thunderx3t110
   thunderxt81
   thunderxt83
   thunderxt88
   tsv110
   xgene1

  error: unknown CPU: 'armv8'
  gmake[2]: *** [CMakeFiles/zlib.dir/build.make:384: CMakeFiles/zlib.dir/arch/arm/insert_string_acle.c.o] Error 1
  info: available CPUs for architecture 'aarch64':
   a64fx
   ampere1
   apple_a10
   apple_a11
   apple_a12
   apple_a13
   apple_a14
   apple_a7
   apple_a8
   apple_a9
   apple_latest
   apple_m1
   apple_s4
   apple_s5
   carmel
   cortex_a34
   cortex_a35
   cortex_a510
   cortex_a53
   cortex_a55
   cortex_a57
   cortex_a65
   cortex_a65ae
   cortex_a710
   cortex_a72
   cortex_a73
   cortex_a75
   cortex_a76
   cortex_a76ae
   cortex_a77
   cortex_a78
   cortex_a78c
   cortex_r82
   cortex_x1
   cortex_x1c
   cortex_x2
   cyclone
   emag
   exynos_m1
   exynos_m2
   exynos_m3
   exynos_m4
   exynos_m5
   falkor
   generic
   kryo
   neoverse_512tvb
   neoverse_e1
   neoverse_n1
   neoverse_n2
   neoverse_v1
   saphira
   thunderx
   thunderx2t99
   thunderx3t110
   thunderxt81
   thunderxt83
   thunderxt88
   tsv110
   xgene1

  error: unknown CPU: 'armv8'
  gmake[2]: *** [CMakeFiles/zlib.dir/build.make:398: CMakeFiles/zlib.dir/arch/arm/adler32_neon.c.o] Error 1
  gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/zlib.dir/all] Error 2
  gmake: *** [Makefile:136: all] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2

  build script failed, must exit now', /home/runner/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cmake-0.1.50/src/lib.rs:1098:5
```

</details>

--- a/src/zig.rs
+++ b/src/zig.rs
@@ -143,12 +143,21 @@ impl Zig {
                     return None;
                 } else if is_riscv64 {
                     return Some("-march=generic_rv64".to_string());
-                } else if arg.starts_with("-march=armv8-a")
-                    && target
+                } else if arg.starts_with("-march=armv8-a") {
+                    if target
                         .map(|x| x.starts_with("aarch64-macos"))
                         .unwrap_or_default()
-                {
-                    return Some(arg.replace("armv8-a", "apple_a14"));
+                    {
+                        return Some(arg.replace("armv8-a", "apple_a14"));
+                    } else if target
+                        .map(|x| x.starts_with("aarch64-linux"))
+                        .unwrap_or_default()
+                    {
+                        return Some(
+                            arg.replace("armv8-a", "generic+v8a")
+                                .replace("simd", "neon"),
+                        );
+                    }
                 }
             }
             Some(arg.to_string())

--- a/tests/hello-cmake/Cargo.toml
+++ b/tests/hello-cmake/Cargo.toml
@@ -6,4 +6,4 @@ edition = "2018"
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-libz-sys = { version = "1.1.8", default-features = false, features = ["zlib-ng"] }
+libz-ng-sys = "1.1.9"

--- a/tests/hello-cmake/src/main.rs
+++ b/tests/hello-cmake/src/main.rs
@@ -1,6 +1,6 @@
 use std::ffi::CStr;
 
-use libz_sys::zlibVersion;
+use libz_ng_sys::zlibVersion;
 
 fn main() {
     let ver = unsafe { zlibVersion() };

